packages/runtime/src/next/client.ts

```diff
@@ -298,48 +291,6 @@ export class Makeswift {
               remaining.push(child)
             }
           })
-
-          if (descriptor.type === ShapeControlType) {
-            const prop = props[propName] as ShapeControlData
-
-            if (prop == null) return
-
-            getResourcesFromElementDescriptors(descriptor.config.type, prop)
```

Comment:
This feels wrong. Shouldn't we be looping through the keys and calling `getResources....` on `prop[key]`?

Reply:
It's _very_ confusing, but it's not wrong. What's happening is that `getElementDescriptors` expects an object of propName -> descriptor and a props object. Since a Shape control keeps a map from shape key -> descriptor then it works out. Another way to think about it is that the "control" for element's props is the shape of the props

Reply:
Got it. That makes sense. It's the same reason we call `getResourcesFromElementDescriptors(elementDescriptors, element.props)` and internally it iterates the props internally.

Reply:
Yeah, exactly. Again, it's very confusing since the function name has _element_ in it even though it operates on arbitrary data, but that's fine. It's going to the farm now...